src/main/java/net/masterthought/cucumber/json/Match.java

```diff
@@ -1,13 +1,55 @@
 package net.masterthought.cucumber.json;
 
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.LinkedList;
+import java.util.List;
+
 public class Match {
 
     // Start: attributes from JSON file report
-    private final String location = null;
+    private final JsonElement location = null;
     // End: attributes from JSON file report
 
     public String getLocation() {
-        return location;
+        if (location==null){
+            return "";
+        }
+        if (location.isJsonPrimitive() && location.getAsJsonPrimitive().isString()) {
+            return location.getAsString();
+        } else {
+
+            String filename = location.getAsJsonObject().get("filepath").getAsJsonObject().get("filename").getAsString();
+
+            LinkedList<Integer> lineList = new LinkedList<>();
```

Comment:
if i change it to List i would have to change lineList.getFirst() and lineList.getLast() to 
lineList.get(0) and lineList.get(lineList.size()-1)  which are less readable